app/controllers/OrganisationDetailsController.scala

```diff
@@ -29,12 +29,41 @@ import scala.concurrent.Future
 trait OrganisationDetailsController extends LisaBaseController {
   val rosmService:RosmService
 
+  def businessLables(businessStructure:  Option[BusinessStructure]): String = {
+    val acceptableValues = Map("Corporate Body" -> "Corporation Tax reference number",
+    "Limited Liability Partnership" -> "Partnership Unique Tax reference number"
+    )
+
+    businessStructure match {
+      case None => {
+        throw new NoBusinessStructureException
+      }
+      case Some(_) => {
+        try {
+          acceptableValues(businessStructure.get.businessStructure)
+        }
+        catch {
+          case e: Exception => throw new Exception("Invalid business structure")
+        }
+      }
+    }
+  }
+
   val get: Action[AnyContent] = Action.async { implicit request =>
     authorisedForLisa { (cacheId) =>
-      cache.fetchAndGetEntry[OrganisationDetails](cacheId, OrganisationDetails.cacheKey).map {
-        case Some(data) => Ok(views.html.registration.organisation_details(OrganisationDetails.form.fill(data)))
-        case None => Ok(views.html.registration.organisation_details(OrganisationDetails.form))
+      cache.fetchAndGetEntry[BusinessStructure](cacheId, BusinessStructure.cacheKey).flatMap { bStructure =>
```

Comment:
As suggested elsewhere I'd suggest doing the Some/None pattern match here - with a None resulting in the Redirect (rather than an Exception recovery)